libs/MeshKernel/include/MeshKernel/BilinearInterpolationOnGriddedSamples.hpp

```diff
@@ -81,21 +82,164 @@ namespace meshkernel
 
         /// @brief Gets the sample value at specific row and column
         /// @return The sample value
-        [[nodiscard]] double getGriddedValue(UInt columnIndex, UInt rowIndex) const
+        [[nodiscard]] values_type getGriddedValue(UInt columnIndex, UInt rowIndex) const
         {
             const auto index = rowIndex * m_numXCoord + columnIndex;
-            return m_values[index];
+            return static_cast<values_type>(m_values[index]);
         }
 
-        const Mesh2D& m_mesh; ///< Reference to the mesh
-        UInt m_numXCoord;     ///< The number of x coordinates of the gridded data
-        UInt m_numYCoord;     ///< The number of y coordinates of the gridded data
-        Point m_origin;       ///< The coordinate of the origin
-        double m_cellSize;    ///< The grid cell size
+        const Mesh2D& m_mesh;    ///< Reference to the mesh
+        UInt m_numXCoord;        ///< The number of x coordinates of the gridded data
+        UInt m_numYCoord;        ///< The number of y coordinates of the gridded data
+        Point m_origin;          ///< The coordinate of the origin
+        double m_cellSize = 0.0; ///< The grid cell size
 
         std::vector<double> m_xCoordinates; ///< The x coordinates of the grid
         std::vector<double> m_yCoordinates; ///< The y coordinates of the grid
-        std::vector<double> m_values;       ///< The gridded sample values
+        const values_type* m_values;        ///< The gridded sample values
```

Comment:
if you end up passing `const std::vector<ValueType>&` to the constructor, m`m_value` should be a const ref. In the original impl, a copy is made.